src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpGatewayControllerBase.cs

```diff
@@ -57,13 +59,24 @@ public string SetupGatewayPageOptionTexts(RoatpGatewayPageViewModel viewModel)
             }
         }
 
-        protected async Task<IActionResult> SubmitGatewayPageAnswer(RoatpGatewayPageViewModel viewModel, string errorViewName)
+        protected async Task<IActionResult> SubmitGatewayPageAnswer(RoatpGatewayPageViewModel viewModel,
+            string errorViewName)
         {
-            var validationResponse = await _gatewayValidator.Validate(viewModel);
+            var validationResponse = await GatewayValidator.Validate(viewModel);
+            return await SubmitGatewayPageAnswer(viewModel, errorViewName, validationResponse.Errors);
+        }
+
+        protected async Task<IActionResult> SubmitGatewayPageAnswer(RoatpGatewayPageViewModel viewModel, string errorViewName,  List<ValidationErrorDetail> validationErrors)
+        {
+            if (validationErrors == null)
+            {    
+                var validationResponse = await GatewayValidator.Validate(viewModel);
+                validationErrors = validationResponse.Errors;
+            }
```

Comment:
I'd remove this block...if they're calling this overload it'd be safe to assume validation has been done.

Reply:
done